gravitee-node-kubernetes/src/main/java/io/gravitee/node/kubernetes/keystoreloader/KubernetesConfigMapKeyStoreLoader.java

```diff
@@ -44,51 +44,46 @@ public class KubernetesConfigMapKeyStoreLoader extends AbstractKubernetesKeyStor
 
     private static final List<String> SUPPORTED_TYPES = Arrays.asList(
         KeyStoreLoader.CERTIFICATE_FORMAT_JKS.toLowerCase(),
-        KeyStoreLoader.CERTIFICATE_FORMAT_PKCS12.toLowerCase(),
-        KeyStoreLoader.CERTIFICATE_FORMAT_PEM_REGISTRY.toLowerCase()
+        KeyStoreLoader.CERTIFICATE_FORMAT_PKCS12.toLowerCase()
     );
 
     private static final Pattern CONFIGMAP_PATTERN = Pattern.compile("^(.*/configmaps/.*)/.*$");
-    private static final Pattern PEM_REGISTRY_CONFIGMAP_PATTERN = Pattern.compile("^(.*/configmaps)/.*$");
 
     public KubernetesConfigMapKeyStoreLoader(KeyStoreLoaderOptions options, KubernetesClient kubernetesClient) {
         super(options, kubernetesClient);
         prepareLocations();
     }
 
     private void prepareLocations() {
-        this.options.getKubernetesLocations()
-            .forEach(location -> {
-                final Matcher matcher = CONFIGMAP_PATTERN.matcher(location);
-                final Matcher pemRegistryMatcher = PEM_REGISTRY_CONFIGMAP_PATTERN.matcher(location);
-                if (matcher.matches()) {
-                    this.resources.put(matcher.group(1), ResourceQuery.<ConfigMap>from(location).build());
-                } else if (pemRegistryMatcher.matches()) {
-                    this.resources.put(location, ResourceQuery.<ConfigMap>from(location).build());
-                } else {
-                    throw new IllegalArgumentException(
-                        "You must specify a data when using configmap (ex: /my-namespace/configmaps/my-configmap/my-keystore or /my-namespace/configmaps/" +
-                        GRAVITEEIO_PEM_REGISTRY +
-                        ")."
-                    );
-                }
-            });
+        if (this.options.getKubernetesLocations() != null && !this.options.getKubernetesLocations().isEmpty()) {
+            this.options.getKubernetesLocations()
+                .forEach(location -> {
+                    final Matcher matcher = CONFIGMAP_PATTERN.matcher(location);
+                    if (matcher.matches()) {
+                        this.resources.put(matcher.group(1), ResourceQuery.<ConfigMap>from(location).build());
+                    } else {
+                        throw new IllegalArgumentException(
+                            "You must specify a data when using configmap (ex: /my-namespace/configmaps/my-configmap/my-keystore)."
+                        );
+                    }
+                });
+        } else if (options.getKeyStoreType().equalsIgnoreCase(CERTIFICATE_FORMAT_PEM_REGISTRY)) {
+            this.resources.put(CERTIFICATE_FORMAT_PEM_REGISTRY, null);
+        }
     }
 
     public static boolean canHandle(KeyStoreLoaderOptions options) {
         final List<String> kubernetesLocations = options.getKubernetesLocations();
 
+        if (options.getKeyStoreType().equalsIgnoreCase(CERTIFICATE_FORMAT_PEM_REGISTRY)) {
```

Comment:
If I understand well now the config looks like this:
```yaml
<server-id>:
  secured: true
  ssl:
    keystore: pem-regsitry
```
or 
```yaml
<server-id>:
  secured: true
  ssl:
    keystore: pkcs12
    kubernetes: 
      - /mynamespace/configmaps/...
```

I suggest you rename pem-regsitry for something more k8s related as the `kubernetes`goes away in that case.

Reply:
It will be the 1st one